aiogram/types/message.py

```diff
@@ -3765,7 +3765,13 @@ def get_url(self, force_private: bool = False) -> Optional[str]:
             else self.chat.username
         )
 
-        return f"https://t.me/{chat_value}/{self.message_id}"
+        message_id_value = (
+            f"{self.message_id}"
+            if not self.message_thread_id
+            else f"{self.message_thread_id}/{self.message_id}"
```

Comment:
What if I don't wanna get in-thread link?
You should save ability to get direct link within merged chat (also to save compatibility)

Reply:
When I tested patch manually, I tested the following cases:

1. Regular chat without topics. The link comes without a topic in the form: `https://t.me/test_chat*/53`

1. Chat with topics. The link comes in the form: `https://t.me/test_chat*/2/54`. 
But at the same time, posts in the default topic (General) come without a thread ID.` https://t.me/test_chat*/55`

1. Chat with topics, but one of the users has disabled the topic view:
 - A user whose topics are disabled always writes to the main group. and the link to his post comes without a topic  `https://t.me/test_chat*/55`.
 - If such a user reply to a message from a topic (it is marked with a topic tag) then the link to his response will contain the topic ID.   `https://t.me/test_chat*/2/54`.  But when this user clicks on this link, the message is found without problems in chat viewed as messages. 
 - The link  `https://t.me/test_chat*/2/54` is valid for both users. And for the one whose chat is viewed as messages and for the one who has a chat with topics.

This behavior seems like in official Telegram client.

Perhaps I haven't checked something or am missing some important idea

Reply:
But I find another bug. Let's this PR still blocking.

So if you reply to another massage, self.message_thread_id suddenly equals the message ID of the replied message.

I back to original image of aiogram lib and made this handler:

```
@router.message()
async def echo_handler(message: Message) -> None:
    url = message.get_url()   
    if url:
        await message.answer(url)
        await message.answer(f"thread_id = {message.message_thread_id}")
    else:
        await message.answer("Ссылка для частных групп и каналов недоступна")
        await message.answer(f"thread_id = {message.message_thread_id}")
    return
```

Reply:
> But I find another bug. 

It's not a bug. Answer inherits thread_id

Reply:
> Perhaps I haven't checked something or am missing some important idea

You miss case with message posted to the thread, but displayed within merged group. Look at my example - different views for the same message.

Reply:
I checked those links again. 

    https://t.me/aiogram/48111/67389
    https://t.me/aiogram/67389

Both in the mobile client and in the web version, when you click on both of these links, the behavior is the same:

I get to the topic and see the message. And the chat looks like with topics. No differences are visible. Even if in the display settings topics are disabled and chat as messages is disabled. 

But. The different behavior if I copy a link to a message. If convert chat back, so links like https://t.me/aiogram/48111/67389  will become broken.

Now I understand that this is a subtle hint at an additional function argument. I'll do.

Reply:
> It's not a bug. Answer inherits thread_id

Please clarify this moment.

It makes my PR useless. Because in the Telegram update we get a thread_id, which in fact is not a chat topic id, but a thread for replying to messages. It would be stupid to override the behavior of the telegram api.

Reply:
I have implemented the following logic:

Added the `include_thread_id` parameter which is False by default.

A thread ID is added to the link only if and  at the same time 
- this parameter `include_thread_id` is true 
- the message is not in the main topic
- the message is not a reply to the another message.